api/rebac/v1/check.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package api.rebac.v1;
+import "google/api/annotations.proto";
+option go_package = "ciam-rebac/api/rebac/v1;v1";
+option java_multiple_files = true;
+option java_package = "api.rebac.v1";
+
+service Check {
+	rpc CheckPermission (CheckPermissionRequest) returns (CheckPermissionResponse) {
```

Comment:
I wonder if we shouldn't brand this more generically to be about checking relationships rather than permissions.

For example, you might check whether someone should receive a notification. (Yes, okay, this needs to be permission-aware, but it's not _only_ about permissions–it's also about whether a user is subscribed.)

CheckRelation?

In which case, I would also consider putting this under the /relationships path, too. I know it's not as simple as that, but I'd like us to think if there is some variant of that that could work.

Reply:
Yeah, so maybe we should change the enum to be more general. REACHABLE vs UNREACHABLE? Just an example.

> Such as:
> Is user:bob a member of the group:bob_club?

IMO there is nothing wrong with using check for this purpose.

On the other hand, if you wanted to refactor this later, it would require a migration of the relation data.

On the _other_ other hand, this kind of thing would be protectable via meta authz rules.

Reply:
Looking at OpenFGA for inspiration here too: https://buf.build/openfga/api/docs/main:openfga.v1#openfga.v1.CheckRequest

I somewhat like that they just deferred a noun all together and went with "check" RPC and "CheckRequest."

Reply:
`rpc CheckPermission` to `rpc CheckRequest`

and keep the endpoint to `check`?

Reply:
As loathe as I am to extend the debate, surely `CheckPermission` is clearer than `CheckRequest`? In the overwhelming majority of cases where permission is the thing being checked, CheckPermission is very clear.

In my opinion, if we really want to do a more generic relation/relationship check, then we could create a second endpoint (preserving the clarity that comes with CheckPermission).